worker/handler/handler.go

```diff
@@ -277,6 +309,16 @@ func (h *Handler) RunStart() (ch *sb.Channel, err error) {
 
 	// create sandbox if needed
 	if h.sandbox == nil {
+		// if memUsage exceeds the sbSoftLimit, we would create at most maxNumSbCreating
+		// sandboxes at the same time; if memUsage exceeds the sbHardLimit, we would block 
+		// all the creation requests until there are more free mem.
+		hms.sbCreatingCond.L.Lock()
+		for (getMemUsage() > hms.sbSoftLimit && hms.numSbCreating > hms.maxNumSbCreating) || getMemUsage() > hms.sbHardLimit {
+			hms.sbCreatingCond.Wait()
```

Comment:
Shouldn't there be some kind of timeout here?  You only broadcast to this cond var in two places:
1. after a container has started
2. after a container has been "nuked"

Even if the set of containers isn't changing, it's possible the existing containers could release some of the memory they've been using, in which case we'd want to start more containers.  Since there's no way for us to be notified when a container starts using less memory, I think on some level we need to rely on polling.

Maybe you can add a timeout for your cond variable?

Reply:
A timeout makes sense, would update.

Reply:
Great, looking forward to seeing this added in the other PR.

Reply:
Updated in https://github.com/open-lambda/open-lambda/pull/54.